src/business_logic/execution/execution_entry_point.rs

```diff
@@ -47,7 +48,7 @@ impl ExecutionEntryPoint {
         caller_address: Address,
         entry_point_type: EntryPointType,
         call_type: Option<CallType>,
-        class_hash: Option<[u8; 32]>,
+        class_hash: [u8; 32],
```

Comment:
I think this is not correct. 
In theory you can run arbitrary cairo programs, no only contracts. In that case you would not have a class hash.

@mmsc2 this is not the case for the tx_context (where you should have that data)

@juanbono what do you think?

Reply:
i thought that it may be the same to use  UNITIALIZED_CLASS_HASH vairable and thus avoid the option type but, semantically they are not the same.